src/react/hooks/useQuery.ts

```diff
@@ -100,360 +163,452 @@ function _useQuery<
-
-          // Do the "unsubscribe" with a short delay.
-          // This way, an existing subscription can be reused without an additional
-          // request if "unsubscribe"  and "resubscribe" to the same ObservableQuery
-          // happen in very fast succession.
-          return () => {
-            setTimeout(() => subscription.unsubscribe());
-            this.forceUpdate = () => this.forceUpdateState();
-          };
-        },
-        [
-          // We memoize the subscribe function using useCallback and the following
-          // dependency keys, because the subscribe function reference is all that
-          // useSyncExternalStore uses internally as a dependency key for the
-          // useEffect ultimately responsible for the subscription, so we are
-          // effectively passing this dependency array to that useEffect buried
-          // inside useSyncExternalStore, as desired.
-          obsQuery,
-          // eslint-disable-next-line react-hooks/exhaustive-deps
-          this.renderPromises,
-          // eslint-disable-next-line react-hooks/exhaustive-deps
-          this.client.disableNetworkFetches,
-        ]
+              },
+              resultData,
+              observable,
+              client,
+              partialRefetch,
+              handleStoreChange,
+              callbackRef.current
+            );
+          }
+        };
+
+        // TODO evaluate if we keep this in
+        // React Compiler cannot handle scoped `let` access, but a mutable object
+        // like this is fine.
+        // was:
+        // let subscription = observable.subscribe(onNext, onError);
+        const subscription = { current: observable.subscribe(onNext, onError) };
+
+        // Do the "unsubscribe" with a short delay.
+        // This way, an existing subscription can be reused without an additional
+        // request if "unsubscribe"  and "resubscribe" to the same ObservableQuery
+        // happen in very fast succession.
+        return () => {
+          setTimeout(() => subscription.current.unsubscribe());
+        };
+      },
+
+      [
+        disableNetworkFetches,
+        skipSubscribing,
+        observable,
+        resultData,
+        partialRefetch,
+        client,
+      ]
+    ),
+    () =>
+      getCurrentResult(
+        resultData,
+        observable,
+        callbackRef.current,
+        partialRefetch,
+        client
       ),
+    () =>
+      getCurrentResult(
+        resultData,
+        observable,
+        callbackRef.current,
+        partialRefetch,
+        client
+      )
+  );
+}
 
-      () => this.getCurrentResult(),
-      () => this.getCurrentResult()
-    );
-
-    // TODO Remove this method when we remove support for options.partialRefetch.
-    this.unsafeHandlePartialRefetch(result);
+function useRegisterSSRObservable<
+  TData = any,
+  TVariables extends OperationVariables = OperationVariables,
+>(
+  observable: ObsQueryWithMeta<TData, TVariables>,
```

Comment:
I don't think these generics really matter here since this doesn't really have any affect on the end-result (i.e. we don't return anything here so we have no way of possibly changing the type). Could we drop the generics here and simplify this?

```suggestion
function useRegisterSSRObservable(
  observable: ObsQueryWithMeta<any, any>,
```